### PR TITLE
refactor(core): initial multiple servers infra

### PR DESCRIPTION
**Pull request by EYHN**
This is the initial refactoring of affine to support multiple servers, but many more changes are needed to make multi-server actually work.

### CODE REVIEW

**Comment by EYHN** [comment]
* **#8813** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8813?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8745** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8745?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8745?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/component/src/components/auth-components/confirm-change-email.tsx [hunk 1]
```diff
@@ -1,22 +0,0 @@
-import { useI18n } from '@affine/i18n';
-import type { FC } from 'react';
-
-import { Button } from '../../ui/button';
-import { AuthPageContainer } from './auth-page-container';
-
-export const ConfirmChangeEmail: FC<{
```
Comment (EYHN) [inline packages/frontend/component/src/components/auth-components/confirm-change-email.tsx:7]: moved to @affine/core

File: packages/frontend/component/src/components/auth-components/email-verified-email.tsx [hunk 2]
```diff
@@ -1,22 +0,0 @@
-import { useI18n } from '@affine/i18n';
-import type { FC } from 'react';
-
-import { Button } from '../../ui/button';
-import { AuthPageContainer } from './auth-page-container';
-
-export const ConfirmVerifiedEmail: FC<{
```
Comment (EYHN) [inline packages/frontend/component/src/components/auth-components/email-verified-email.tsx:7]: moved to affine/core

File: packages/frontend/core/src/components/affine/auth/oauth.tsx [hunk 3]
```diff
@@ -29,11 +28,11 @@ const OAuthProviderMap: Record<
...
 
 export function OAuth({ redirectUrl }: { redirectUrl?: string }) {
-  const serverConfig = useService(ServerConfigService).serverConfig;
+  const serverService = useService(ServerService);
   const urlService = useService(UrlService);
-  const oauth = useLiveData(serverConfig.features$.map(r => r?.oauth));
+  const oauth = useLiveData(serverService.server.features$.map(r => r?.oauth));
   const oauthProviders = useLiveData(
-    serverConfig.config$.map(r => r?.oauthProviders)
+    serverService.server.config$.map(r => r?.oauthProviders)
```
Comment (forehalo) [inline packages/frontend/core/src/components/affine/auth/oauth.tsx:36]: ServerService heard weird, renamed to CloudService?

Comment (EYHN) [reply packages/frontend/core/src/components/affine/auth/oauth.tsx:36]: Many places use "server", serverConfig, serverList, so server is used here for consistency.

File: packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        .gql({
+          query: changeEmailMutation,
+          variables: {
+            token: token,
+            email: email,
+          },
+        })
+        .catch(err => {
+          if (err instanceof BackendError) {
+            const userFriendlyError = UserFriendlyError.fromAnyError(err);
```
Comment (forehalo) [inline packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx]: originError in BackendError is UserFriendlyError already

Comment (forehalo) [reply packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx]: you can update the implementation of `fromAnyError` to extract it from BackendError

File: packages/frontend/core/src/modules/cloud/services/global-server.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { ServerDeploymentType } from '@affine/graphql';
+import { Service } from '@toeverything/infra';
+
+import type { Server } from '../entities/server';
+import type { ServersService } from './servers';
+
+export class GlobalServerService extends Service {
```
Comment (forehalo) [inline packages/frontend/core/src/modules/cloud/services/global-server.ts]: `DefaultServerService` ?

File: packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            if (userFriendlyError.name === ErrorNames.INVALID_EMAIL_TOKEN) {
+              return navigateHelper.jumpToExpired();
+            }
+          }
+          throw err;
+        });
+      setIsLoading(true);
```
Comment (pengx17) [inline packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx]: Seems to be in the wrong place. Move to line 30?

File: packages/frontend/core/src/desktop/pages/auth/email-verified-email.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+          },
+        })
+        .catch(error => {
+          const userFriendlyError = UserFriendlyError.fromAnyError(error);
+          if (userFriendlyError.name === ErrorNames.INVALID_EMAIL_TOKEN) {
+            return navigateHelper.jumpToExpired();
+          }
+          throw error;
+        });
+      setIsLoading(true);
```
Comment (pengx17) [inline packages/frontend/core/src/desktop/pages/auth/email-verified-email.tsx]:
```suggestion
      const token = searchParams.get('token') ?? '';
      setIsLoading(true);
      await graphqlService
        .gql({
          query: verifyEmailMutation,
          variables: {
            token: token,
          },
        })
        .catch(error => {
          const userFriendlyError = UserFriendlyError.fromAnyError(error);
          if (userFriendlyError.name === ErrorNames.INVALID_EMAIL_TOKEN) {
            return navigateHelper.jumpToExpired();
          }
          throw error;
        });
```

File: packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx (hunk 4 above)
Comment (EYHN) [reply packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx]: we already have 'originError' check in `fromAnyError`

Comment (forehalo) [reply packages/frontend/core/src/desktop/pages/auth/confirm-change-email.tsx]: if so, instanceof is not necessary

**Review by pengx17 — APPROVED** [review]

**Pull request merged**